cmake/SetupCodeChecks.cmake

```diff
@@ -99,15 +104,22 @@ macro(blt_add_uncrustify_check)
 
     ## parse the arguments to the macro
     set(options)
-    set(singleValueArgs CFG_FILE)
-    set(multiValueArgs SRC_FILES)
+    set(singleValueArgs CFG_FILE COMMENT WORKING_DIRECTORY)
+    set(multiValueArgs SRC_FILES FLAGS REDIRECT)
     cmake_parse_arguments(arg
         "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
 
+    if(${arg_WORKING_DIRECTORY})
+        set(_wd ${arg_WORKING_DIRECTORY})
+    else()
+        set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
+    endif()
+
     add_custom_target("uncrustify_check_${PROJECT_NAME}"
```

Comment:
It would be nice if there were an option to set the name of this target rather than using the cmake project.